From 04178918e7f6b5f34dde81ec79ee8a1ccace3be3 Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Mon, 17 Oct 2022 11:45:03 +0200
Subject: [PATCH] wifi: ath11k: pci: fix compilation in 5.16 and older

Commit ("genirq/msi, treewide: Use a named struct for PCI/MSI attributes")
changed the msi_desc structure a bit, however that is only available in
kernels 5.17 and newer, so check for kernel version to allow compilation
in 5.16 and older.

Signed-off-by: Robert Marko <robimarko@gmail.com>
---
 drivers/net/wireless/ath/ath11k/pci.c | 4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/net/wireless/ath/ath11k/pci.c
+++ b/drivers/net/wireless/ath/ath11k/pci.c
@@ -459,7 +459,11 @@ static int ath11k_pci_alloc_msi(struct a
 	pci_read_config_dword(pci_dev, pci_dev->msi_cap + PCI_MSI_ADDRESS_LO,
 			      &ab->pci.msi.addr_lo);
 
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(5, 17, 0))
 	if (msi_desc->pci.msi_attrib.is_64) {
+#else
+	if (msi_desc->msi_attrib.is_64) {
+#endif
 		pci_read_config_dword(pci_dev, pci_dev->msi_cap + PCI_MSI_ADDRESS_HI,
 				      &ab->pci.msi.addr_hi);
 	} else {
